src/convertToObject.js

```diff
@@ -6,7 +6,27 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const splited = sourceString.split(';');
+
+  for (const item of splited) {
+    if (item.trim() === '') {
+      continue;
+    }
+
+    const [prop, value] = item.split(':');
+
+    if (!prop || !value) {
+      continue;
+    }
+
+    const trimmedProp = prop.trim();
+    const trimmedValue = value.trim();
+
+    result[trimmedProp] = trimmedValue;
+  }
```

Comment:
Instead of using a 'for...of' loop, consider using array methods like 'filter' to remove empty items and 'map' to transform each item into a [key, value] pair. This approach aligns with the checklist item about avoiding loops and using iteration methods.